cdk-infra/lib/constructs/auto-builder-api-construct.ts

```diff
@@ -0,0 +1,151 @@
+import { Cors, CorsOptions, LambdaIntegration, LambdaRestApi } from 'aws-cdk-lib/aws-apigateway';
+import { Code, Function, Runtime } from 'aws-cdk-lib/aws-lambda';
+import { BundlingOptions, NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import { IQueue } from 'aws-cdk-lib/aws-sqs';
+import { StringParameter } from 'aws-cdk-lib/aws-ssm';
+import { Construct } from 'constructs';
+
+import path = require('path');
+
+const HANDLERS_PATH = path.join(__dirname, '/../../../api/controllers/v2/');
+
+const bundling: BundlingOptions = {
+  sourceMap: true,
+  minify: true,
+  loader: {
+    '.node': 'file',
+  },
+  commandHooks: {
+    beforeBundling: (inputDir: string, outputDir: string): string[] => {
+      return [`cp -a ${inputDir}/static/api/config ${outputDir}`];
+    },
+    afterBundling: (): string[] => [],
+    beforeInstall: (): string[] => [],
+  },
+};
+
+const runtime = Runtime.NODEJS_18_X;
+
+interface AutoBuilderApiConstructProps {
+  jobsQueue: IQueue;
+  jobUpdatesQueue: IQueue;
+  environment: Record<string, string>;
+}
+
+export class AutoBuilderApiConstruct extends Construct {
+  constructor(scope: Construct, id: string, props: AutoBuilderApiConstructProps) {
+    super(scope, id);
+
+    const { jobsQueue, jobUpdatesQueue, environment } = props;
+
+    const slackTriggerLambda = new NodejsFunction(this, 'slackTriggerLambda', {
+      entry: `${HANDLERS_PATH}/slack.ts`,
+      runtime,
+      handler: 'DeployRepo',
+      environment,
+      bundling,
+    });
+
+    const slackDisplayRepoLambda = new NodejsFunction(this, 'slackDisplayRepoLambda', {
+      entry: `${HANDLERS_PATH}/slack.ts`,
+      runtime,
+      handler: 'DeployRepoDisplayRepoOptions',
+      environment,
+      bundling,
+    });
+
+    const fastlyDochubToken = StringParameter.fromSecureStringParameterAttributes(this, 'fastlyDochubToken', {
+      parameterName: '/env/dev/docs/worker_pool/fastly/docs/dochub/token',
```

Comment:
Are we intentionally grabbing this from the `dev` env for now, or do we want to grab this per env?

Reply:
Yep! Intentionally grabbing dev for now.